workloads/hyper-scale/common.sh

```diff
@@ -0,0 +1,98 @@
+        unzip awscliv2.zip
+        ./aws/install
+    fi
+}
+setup(){
+    export MGMT_CLUSTER_NAME=$(oc get infrastructure cluster -o jsonpath='{.status.infrastructureName}'| cut -c 1-13)
+    export BASEDOMAIN=$(oc get dns cluster -o jsonpath='{.spec.baseDomain}')
+    echo [default] > aws_credentials
+    echo aws_access_key_id=$AWS_ACCESS_KEY_ID >> aws_credentials
+    echo aws_secret_access_key=$AWS_SECRET_ACCESS_KEY >> aws_credentials
+    rosa login --env=${ROSA_ENVIRONMENT}
+    ocm login --url=https://api.stage.openshift.com --token="${ROSA_TOKEN}"
+    rosa whoami
+    rosa verify quota
+    rosa verify permissions
+}
+
+install(){
+    echo "Create S3 bucket and route53 doamin.."
+    aws s3api create-bucket --acl public-read --bucket $MGMT_CLUSTER_NAME-aws-rhperfscale-org --create-bucket-configuration LocationConstraint=$AWS_REGION --region $AWS_REGION || true
+    aws route53 create-hosted-zone --name $BASEDOMAIN --caller-reference perfscale-ci-$(date --iso-8601=seconds) || true
+    echo "Wait till S3 bucket is ready.."
+    aws s3api wait bucket-exists --bucket $MGMT_CLUSTER_NAME-aws-rhperfscale-org 
+    hypershift install --oidc-storage-provider-s3-bucket-name $MGMT_CLUSTER_NAME-aws-rhperfscale-org --oidc-storage-provider-s3-credentials aws_credentials --oidc-storage-provider-s3-region $AWS_REGION  --enable-ocp-cluster-monitoring
+    echo "Wait till Operator is ready.."
+    cm=""
+    while [[ $cm != "oidc-storage-provider-s3-config" ]]
+    do
+        cm=$(oc get configmap -n kube-public oidc-storage-provider-s3-config --no-headers | awk '{print$1}' || true)
+        echo "Hypershift Operator is not ready yet..Retrying after few seconds"
+        sleep 5
+    done
+}
+
+create_cluster(){
+    echo $PULL_SECRET > pull-secret
+    hypershift create cluster aws --name $HOSTED_CLUSTER_NAME --node-pool-replicas=$COMPUTE_WORKERS_NUMBER --base-domain $BASEDOMAIN --pull-secret pull-secret --aws-creds aws_credentials --region $AWS_REGION --control-plane-availability-policy $REPLICA_TYPE --network-type $NETWORK_TYPE --instance-type $COMPUTE_WORKERS_TYPE # --control-plane-operator-image=quay.io/hypershift/hypershift:latest
+    echo "Wait till hosted cluster got created and in progress.."
+    oc wait --for=condition=available=false --timeout=60s hostedcluster -n clusters $HOSTED_CLUSTER_NAME
+    oc get hostedcluster -n clusters $HOSTED_CLUSTER_NAME
+}
+
+create_empty_cluster(){
+    echo $PULL_SECRET > pull-secret
+    hypershift create cluster none --name $HOSTED_CLUSTER_NAME --node-pool-replicas=0 --base-domain $BASEDOMAIN --pull-secret pull-secret --control-plane-availability-policy $REPLICA_TYPE --network-type $NETWORK_TYPE
+    echo "Wait till hosted cluster got created and in progress.."
+    oc wait --for=condition=available=false --timeout=60s hostedcluster -n clusters $HOSTED_CLUSTER_NAME
+    oc get hostedcluster -n clusters $HOSTED_CLUSTER_NAME
+}
+
+postinstall(){
```

Comment:
With our tests we generally need to make an adjustment to firewall rules for certain network tests to be able to run post-install. We should look at add those, potentially here, as well. Example from airflow rosa install https://github.com/cloud-bulldozer/airflow-kubernetes/blob/master/dags/openshift_nightlies/scripts/utils/rosa_post_install.py

Reply:
yes, I have an issue created for that on airflow repo, https://github.com/cloud-bulldozer/airflow-kubernetes/issues/160. 
Once that is ready, I will include here as well in a separate PR.

Reply:
sounds good